### ISSUE TITLE
Crash when trying to send coins

### ISSUE THREAD

**Issue by f8192** [HashEngineering/dash-wallet#161]
Hello! As the title says, the wallet does not work. I sent a crash report to hashengineeringsolutions@gmail.com on 22 Dec, but still no fix or any reply. Please fix. Did you receive the report? @HashEngineering

**Comment by HashEngineering** [comment]
Yes, your report was received.  Our next update should address your issue in two ways:
1) Crash reports and other issues will be emailed to support@dash.org where problems and questions will be addressed within 24 hours
2)  Exchange rates will be acquired through multiple sources if the first one fails.

The cause of the crashing was that the exchange rate server BitcoinAverage.com could not be contacted and so app crashed when trying to access a non-existent exchange rate.  Our fixes in version 6.3.0 should resolve this.

```
07:36:09 [AsyncTask #2] ExchangeRatesProvider - problem fetching exchange rates from https://apiv2.bitcoinaverage.com/indices/global/ticker/short?crypto=BTC
java.net.ConnectException: Failed to connect to apiv2.bitcoinaverage.com/159.203.144.149:443
	at com.squareup.okhttp.internal.io.RealConnection.connectSocket(RealConnection.java:143) ~[na:0.0]
	at com.squareup.okhttp.internal.io.RealConnection.connect(RealConnection.java:112) ~[na:0.0]
	at com.squareup.okhttp.internal.http.StreamAllocation.findConnection(StreamAllocation.java:184) ~[na:0.0]
	at com.squareup.okhttp.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:126) ~[na:0.0]
	at com.squareup.okhttp.internal.http.StreamAllocation.newStream(StreamAllocation.java:95) ~[na:0.0]
	at com.squareup.okhttp.internal.http.HttpEngine.connect(HttpEngine.java:281) ~[na:0.0]
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:224) ~[na:0.0]
	at com.squareup.okhttp.Call.getResponse(Call.java:286) ~[na:0.0]
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:243) ~[na:0.0]
	at com.squareup.okhttp.logging.HttpLoggingInterceptor.intercept(HttpLoggingInterceptor.java:204) ~[na:0.0]
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:232) ~[na:0.0]
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:205) ~[na:0.0]
	at com.squareup.okhttp.Call.execute(Call.java:80) ~[na:0.0]
	at de.schildbach.wallet.data.ExchangeRatesProvider.requestExchangeRates(ExchangeRatesProvider.java:264) ~[na:0.0]
	at de.schildbach.wallet.data.ExchangeRatesProvider.query(ExchangeRatesProvider.java:134) ~[na:0.0]
	at android.content.ContentProvider.query(ContentProvider.java:1055) ~[na:0.0]
	at android.content.ContentProvider.query(ContentProvider.java:1147) ~[na:0.0]
	at android.content.ContentProvider$Transport.query(ContentProvider.java:240) ~[na:0.0]
	at android.content.ContentResolver.query(ContentResolver.java:754) ~[na:0.0]
	at android.content.ContentResolver.query(ContentResolver.java:704) ~[na:0.0]
	at android.content.CursorLoader.loadInBackground(CursorLoader.java:64) ~[na:0.0]
	at android.content.CursorLoader.loadInBackground(CursorLoader.java:54) ~[na:0.0]
	at android.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:315) ~[na:0.0]
	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:69) ~[na:0.0]
	at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:64) ~[na:0.0]
	at android.os.AsyncTask$2.call(AsyncTask.java:333) ~[na:0.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:0.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162) ~[na:0.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636) ~[na:0.0]
	at java.lang.Thread.run(Thread.java:764) ~[na:0.0]

```

```
07:36:07 [backgroundThread] CrashReporter - crashing because of uncaught exception
java.lang.NullPointerException: Attempt to read from field 'org.bitcoinj.utils.ExchangeRate org.dash.wallet.common.data.ExchangeRate.rate' on a null object reference
	at de.schildbach.wallet.service.BlockchainServiceImpl$1.onCoinsReceived(BlockchainServiceImpl.java:176) ~[na:0.0]
	at org.bitcoinj.wallet.Wallet$5.run(Wallet.java:2813) ~[na:0.0]
	at org.bitcoinj.utils.Threading$2.execute(Threading.java:142) ~[na:0.0]
	at org.bitcoinj.wallet.Wallet.queueOnCoinsReceived(Wallet.java:2810) ~[na:0.0]
	at org.bitcoinj.wallet.Wallet.maybeCommitTx(Wallet.java:2513) ~[na:0.0]
	at org.bitcoinj.wallet.Wallet.commitTx(Wallet.java:2551) ~[na:0.0]
	at org.bitcoinj.wallet.Wallet.sendCoinsOffline(Wallet.java:3826) ~[na:0.0]
	at de.schildbach.wallet.ui.send.SendCoinsOfflineTask$1.run(SendCoinsOfflineTask.java:61) ~[na:0.0]
	at android.os.Handler.handleCallback(Handler.java:790) ~[na:0.0]
	at android.os.Handler.dispatchMessage(Handler.java:99) ~[na:0.0]
	at android.os.Looper.loop(Looper.java:164) ~[na:0.0]
	at android.os.HandlerThread.run(HandlerThread.java:65) ~[na:0.0]
```

**Comment by HashEngineering** [comment]
6 users in the past 60 days have experienced this resulting in 32 crashes on Android 7, 7.1, 8.0 and 8.1 devices with app version 6.2.1.

**Comment by HashEngineering** [comment]
This was resolved in the "New Price Sourcing" update 6.3.0 - https://github.com/HashEngineering/dash-wallet/commit/6286c6c2841032873e2c67eded0e20ef5eeda742
